diff --git a/build.rs b/build.rs
index 199a9ac..577d20a 100644
--- a/build.rs
+++ b/build.rs
@@ -30,6 +30,7 @@ const ALLOWED_CFGS: &[&str] = &[
     "musl_v1_2_3",
     // Corresponds to `_REDIR_TIME64` in musl
     "musl32_time64",
+    "musl_redir_time64",
     "vxworks_lt_25_09",
 ];
 
@@ -111,15 +112,19 @@ fn main() {
     let musl = target_env == "musl" || target_env == "ohos";
 
     // loongarch64 and ohos only exist with recent musl
-    if target_arch == "loongarch64" || target_env == "ohos" {
+    if target_arch == "wasm32" || target_arch == "loongarch64" || target_env == "ohos" {
         musl_v1_2_3 = true;
     }
 
     if musl && musl_v1_2_3 {
         set_cfg("musl_v1_2_3");
+        if target_arch == "wasm32" {
+            set_cfg("musl32_time64");
+        }
         if MUSL_REDIR_TIME64_ARCHES.contains(&target_arch.as_str()) {
             set_cfg("musl32_time64");
             set_cfg("linux_time_bits64");
+            set_cfg("musl_redir_time64");
         }
     }
 
diff --git a/src/new/common/posix/pthread.rs b/src/new/common/posix/pthread.rs
index 146cd60..17d3906 100644
--- a/src/new/common/posix/pthread.rs
+++ b/src/new/common/posix/pthread.rs
@@ -195,7 +195,7 @@ extern "C" {
 
     #[cfg(any(target_os = "android", target_os = "l4re", target_os = "linux"))]
     #[cfg_attr(gnu_time_bits64, link_name = "__pthread_mutex_timedlock64")]
-    #[cfg_attr(musl32_time64, link_name = "__pthread_mutex_timedlock_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__pthread_mutex_timedlock_time64")]
     pub fn pthread_mutex_timedlock(
         lock: *mut crate::pthread_mutex_t,
         abstime: *const crate::timespec,
diff --git a/src/new/musl/sys/socket.rs b/src/new/musl/sys/socket.rs
index e723043..7d3f495 100644
--- a/src/new/musl/sys/socket.rs
+++ b/src/new/musl/sys/socket.rs
@@ -41,7 +41,7 @@ extern "C" {
         vlen: c_uint,
         flags: c_uint,
     ) -> c_int;
-    #[cfg_attr(musl32_time64, link_name = "__recvmmsg_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__recvmmsg_time64")]
     pub fn recvmmsg(
         sockfd: c_int,
         msgvec: *mut crate::mmsghdr,
diff --git a/src/unix/linux_like/linux/mod.rs b/src/unix/linux_like/linux/mod.rs
index 13d25bd..8508ef4 100644
--- a/src/unix/linux_like/linux/mod.rs
+++ b/src/unix/linux_like/linux/mod.rs
@@ -4112,7 +4112,7 @@ extern "C" {
     pub fn lcong48(p: *mut c_ushort);
 
     #[cfg_attr(gnu_time_bits64, link_name = "__lutimes64")]
-    #[cfg_attr(musl32_time64, link_name = "__lutimes_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__lutimes_time64")]
     pub fn lutimes(file: *const c_char, times: *const crate::timeval) -> c_int;
 
     pub fn shm_open(name: *const c_char, oflag: c_int, mode: mode_t) -> c_int;
@@ -4188,9 +4188,9 @@ extern "C" {
     pub fn fremovexattr(filedes: c_int, name: *const c_char) -> c_int;
     pub fn signalfd(fd: c_int, mask: *const crate::sigset_t, flags: c_int) -> c_int;
     pub fn timerfd_create(clockid: crate::clockid_t, flags: c_int) -> c_int;
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__timerfd_gettime64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__timerfd_gettime64")]
     pub fn timerfd_gettime(fd: c_int, curr_value: *mut crate::itimerspec) -> c_int;
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__timerfd_settime64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__timerfd_settime64")]
     pub fn timerfd_settime(
         fd: c_int,
         flags: c_int,
@@ -4207,7 +4207,7 @@ extern "C" {
     ) -> c_int;
     pub fn dup3(oldfd: c_int, newfd: c_int, flags: c_int) -> c_int;
     #[cfg_attr(gnu_time_bits64, link_name = "__sigtimedwait64")]
-    #[cfg_attr(musl32_time64, link_name = "__sigtimedwait_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__sigtimedwait_time64")]
     pub fn sigtimedwait(
         set: *const sigset_t,
         info: *mut siginfo_t,
@@ -4270,7 +4270,7 @@ extern "C" {
     pub fn eventfd_write(fd: c_int, value: eventfd_t) -> c_int;
 
     #[cfg_attr(gnu_time_bits64, link_name = "__sched_rr_get_interval64")]
-    #[cfg_attr(musl32_time64, link_name = "__sched_rr_get_interval_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__sched_rr_get_interval_time64")]
     pub fn sched_rr_get_interval(pid: crate::pid_t, tp: *mut crate::timespec) -> c_int;
     pub fn sched_setparam(pid: crate::pid_t, param: *const crate::sched_param) -> c_int;
     pub fn setns(fd: c_int, nstype: c_int) -> c_int;
diff --git a/src/unix/linux_like/linux/musl/mod.rs b/src/unix/linux_like/linux/musl/mod.rs
index 59fe8a4..d783857 100644
--- a/src/unix/linux_like/linux/musl/mod.rs
+++ b/src/unix/linux_like/linux/musl/mod.rs
@@ -765,7 +765,7 @@ extern "C" {
         new_limit: *const crate::rlimit,
         old_limit: *mut crate::rlimit,
     ) -> c_int;
-    #[cfg_attr(musl32_time64, link_name = "__gettimeofday_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__gettimeofday_time64")]
     pub fn gettimeofday(tp: *mut crate::timeval, tz: *mut c_void) -> c_int;
     pub fn ptrace(request: c_int, ...) -> c_long;
     pub fn getpriority(which: c_int, who: crate::id_t) -> c_int;
@@ -801,9 +801,9 @@ extern "C" {
     // Added in `musl` 1.2.2
     pub fn reallocarray(ptr: *mut c_void, nmemb: size_t, size: size_t) -> *mut c_void;
 
-    #[cfg_attr(musl32_time64, link_name = "__adjtimex_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__adjtimex_time64")]
     pub fn adjtimex(buf: *mut crate::timex) -> c_int;
-    #[cfg_attr(musl32_time64, link_name = "__clock_adjtime64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__clock_adjtime64")]
     pub fn clock_adjtime(clk_id: crate::clockid_t, buf: *mut crate::timex) -> c_int;
 
     pub fn ctermid(s: *mut c_char) -> *mut c_char;
diff --git a/src/unix/linux_like/linux_l4re_shared.rs b/src/unix/linux_like/linux_l4re_shared.rs
index d91240a..e03b388 100644
--- a/src/unix/linux_like/linux_l4re_shared.rs
+++ b/src/unix/linux_like/linux_l4re_shared.rs
@@ -1725,7 +1725,7 @@ cfg_if! {
                 flags: c_ulong,
             ) -> isize;
             #[cfg_attr(gnu_time_bits64, link_name = "__futimes64")]
-            #[cfg_attr(musl32_time64, link_name = "__futimes_time64")]
+            #[cfg_attr(musl_redir_time64, link_name = "__futimes_time64")]
             pub fn futimes(fd: c_int, times: *const crate::timeval) -> c_int;
         }
     }
@@ -1827,10 +1827,10 @@ extern "C" {
     ) -> c_int;
     pub fn sched_get_priority_max(policy: c_int) -> c_int;
     #[cfg_attr(gnu_time_bits64, link_name = "__settimeofday64")]
-    #[cfg_attr(musl32_time64, link_name = "__settimeofday_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__settimeofday_time64")]
     pub fn settimeofday(tv: *const crate::timeval, tz: *const crate::timezone) -> c_int;
     #[cfg_attr(gnu_time_bits64, link_name = "__sem_timedwait64")]
-    #[cfg_attr(musl32_time64, link_name = "__sem_timedwait_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__sem_timedwait_time64")]
     pub fn sem_timedwait(sem: *mut crate::sem_t, abstime: *const crate::timespec) -> c_int;
     pub fn sem_getvalue(sem: *mut crate::sem_t, sval: *mut c_int) -> c_int;
     pub fn mount(
@@ -1843,7 +1843,7 @@ extern "C" {
     #[cfg_attr(gnu_time_bits64, link_name = "__prctl_time64")]
     pub fn prctl(option: c_int, ...) -> c_int;
     #[cfg_attr(gnu_time_bits64, link_name = "__ppoll64")]
-    #[cfg_attr(musl32_time64, link_name = "__ppoll_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__ppoll_time64")]
     pub fn ppoll(
         fds: *mut crate::pollfd,
         nfds: crate::nfds_t,
@@ -1934,9 +1934,9 @@ extern "C" {
     pub fn timer_delete(timerid: crate::timer_t) -> c_int;
     #[cfg(not(target_os = "l4re"))]
     pub fn timer_getoverrun(timerid: crate::timer_t) -> c_int;
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__timer_gettime64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__timer_gettime64")]
     pub fn timer_gettime(timerid: crate::timer_t, curr_value: *mut crate::itimerspec) -> c_int;
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__timer_settime64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__timer_settime64")]
     pub fn timer_settime(
         timerid: crate::timer_t,
         flags: c_int,
diff --git a/src/unix/linux_like/mod.rs b/src/unix/linux_like/mod.rs
index 3ded4da..f79bb54 100644
--- a/src/unix/linux_like/mod.rs
+++ b/src/unix/linux_like/mod.rs
@@ -1873,20 +1873,20 @@ extern "C" {
     pub fn mincore(addr: *mut c_void, len: size_t, vec: *mut c_uchar) -> c_int;
 
     #[cfg_attr(gnu_time_bits64, link_name = "__clock_getres64")]
-    #[cfg_attr(musl32_time64, link_name = "__clock_getres_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__clock_getres_time64")]
     pub fn clock_getres(clk_id: crate::clockid_t, tp: *mut crate::timespec) -> c_int;
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__clock_gettime64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__clock_gettime64")]
     pub fn clock_gettime(clk_id: crate::clockid_t, tp: *mut crate::timespec) -> c_int;
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__clock_settime64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__clock_settime64")]
     pub fn clock_settime(clk_id: crate::clockid_t, tp: *const crate::timespec) -> c_int;
     #[cfg(not(target_os = "l4re"))]
     pub fn clock_getcpuclockid(pid: crate::pid_t, clk_id: *mut crate::clockid_t) -> c_int;
 
     #[cfg_attr(gnu_time_bits64, link_name = "__getitimer64")]
-    #[cfg_attr(musl32_time64, link_name = "__getitimer_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__getitimer_time64")]
     pub fn getitimer(which: c_int, curr_value: *mut crate::itimerval) -> c_int;
     #[cfg_attr(gnu_time_bits64, link_name = "__setitimer64")]
-    #[cfg_attr(musl32_time64, link_name = "__setitimer_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__setitimer_time64")]
     pub fn setitimer(
         which: c_int,
         new_value: *const crate::itimerval,
@@ -1907,11 +1907,11 @@ extern "C" {
     #[cfg_attr(gnu_file_offset_bits64, link_name = "posix_fadvise64")]
     pub fn posix_fadvise(fd: c_int, offset: off_t, len: off_t, advise: c_int) -> c_int;
     #[cfg_attr(gnu_time_bits64, link_name = "__futimens64")]
-    #[cfg_attr(musl32_time64, link_name = "__futimens_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__futimens_time64")]
     #[cfg(not(target_os = "l4re"))]
     pub fn futimens(fd: c_int, times: *const crate::timespec) -> c_int;
     #[cfg_attr(gnu_time_bits64, link_name = "__utimensat64")]
-    #[cfg_attr(musl32_time64, link_name = "__utimensat_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__utimensat_time64")]
     pub fn utimensat(
         dirfd: c_int,
         path: *const c_char,
@@ -1963,7 +1963,7 @@ extern "C" {
     pub fn setresgid(rgid: crate::gid_t, egid: crate::gid_t, sgid: crate::gid_t) -> c_int;
     #[cfg(not(target_os = "l4re"))]
     pub fn setresuid(ruid: crate::uid_t, euid: crate::uid_t, suid: crate::uid_t) -> c_int;
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__wait4_time64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__wait4_time64")]
     #[cfg(not(target_os = "l4re"))]
     pub fn wait4(
         pid: crate::pid_t,
diff --git a/src/unix/mod.rs b/src/unix/mod.rs
index ac411f1..79bea0a 100644
--- a/src/unix/mod.rs
+++ b/src/unix/mod.rs
@@ -888,7 +888,7 @@ extern "C" {
         all(not(gnu_time_bits64), gnu_file_offset_bits64),
         link_name = "fstat64"
     )]
-    #[cfg_attr(musl32_time64, link_name = "__fstat_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__fstat_time64")]
     pub fn fstat(fildes: c_int, buf: *mut stat) -> c_int;
 
     pub fn mkdir(path: *const c_char, mode: mode_t) -> c_int;
@@ -907,7 +907,7 @@ extern "C" {
         all(not(gnu_time_bits64), gnu_file_offset_bits64),
         link_name = "stat64"
     )]
-    #[cfg_attr(musl32_time64, link_name = "__stat_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__stat_time64")]
     pub fn stat(path: *const c_char, buf: *mut stat) -> c_int;
 
     pub fn pclose(stream: *mut crate::FILE) -> c_int;
@@ -1002,7 +1002,7 @@ extern "C" {
         link_name = "fstatat64"
     )]
     #[cfg(not(target_os = "l4re"))]
-    #[cfg_attr(musl32_time64, link_name = "__fstatat_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__fstatat_time64")]
     pub fn fstatat(dirfd: c_int, pathname: *const c_char, buf: *mut stat, flags: c_int) -> c_int;
     #[cfg(not(target_os = "l4re"))]
     pub fn linkat(
@@ -1112,7 +1112,7 @@ extern "C" {
     )]
     #[cfg_attr(target_os = "netbsd", link_name = "__nanosleep50")]
     #[cfg_attr(gnu_time_bits64, link_name = "__nanosleep64")]
-    #[cfg_attr(musl32_time64, link_name = "__nanosleep_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__nanosleep_time64")]
     pub fn nanosleep(rqtp: *const timespec, rmtp: *mut timespec) -> c_int;
     pub fn tcgetpgrp(fd: c_int) -> pid_t;
     pub fn tcsetpgrp(fd: c_int, pgrp: crate::pid_t) -> c_int;
@@ -1157,7 +1157,7 @@ extern "C" {
     pub fn umask(mask: mode_t) -> mode_t;
 
     #[cfg_attr(target_os = "netbsd", link_name = "__utime50")]
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__utime64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__utime64")]
     pub fn utime(file: *const c_char, buf: *const utimbuf) -> c_int;
 
     #[cfg_attr(
@@ -1212,7 +1212,7 @@ extern "C" {
         all(not(gnu_time_bits64), gnu_file_offset_bits64),
         link_name = "lstat64"
     )]
-    #[cfg_attr(musl32_time64, link_name = "__lstat_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__lstat_time64")]
     pub fn lstat(path: *const c_char, buf: *mut stat) -> c_int;
 
     #[cfg_attr(
@@ -1244,7 +1244,7 @@ extern "C" {
 
     #[cfg_attr(target_os = "netbsd", link_name = "__getrusage50")]
     #[cfg_attr(gnu_time_bits64, link_name = "__getrusage64")]
-    #[cfg_attr(musl32_time64, link_name = "__getrusage_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__getrusage_time64")]
     pub fn getrusage(resource: c_int, usage: *mut rusage) -> c_int;
 
     #[cfg_attr(
@@ -1327,7 +1327,7 @@ extern "C" {
         link_name = "pthread_cond_timedwait$UNIX2003"
     )]
     #[cfg_attr(gnu_time_bits64, link_name = "__pthread_cond_timedwait64")]
-    #[cfg_attr(musl32_time64, link_name = "__pthread_cond_timedwait_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__pthread_cond_timedwait_time64")]
     pub fn pthread_cond_timedwait(
         cond: *mut crate::pthread_cond_t,
         lock: *mut crate::pthread_mutex_t,
@@ -1396,11 +1396,11 @@ extern "C" {
 
     #[cfg_attr(target_os = "netbsd", link_name = "__utimes50")]
     #[cfg_attr(gnu_time_bits64, link_name = "__utimes64")]
-    #[cfg_attr(musl32_time64, link_name = "__utimes_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__utimes_time64")]
     pub fn utimes(filename: *const c_char, times: *const crate::timeval) -> c_int;
     pub fn dlopen(filename: *const c_char, flag: c_int) -> *mut c_void;
     pub fn dlerror() -> *mut c_char;
-    #[cfg_attr(musl32_time64, link_name = "__dlsym_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__dlsym_time64")]
     pub fn dlsym(handle: *mut c_void, symbol: *const c_char) -> *mut c_void;
     pub fn dlclose(handle: *mut c_void) -> c_int;
 
@@ -1451,42 +1451,42 @@ extern "C" {
     #[cfg_attr(target_os = "netbsd", link_name = "__gmtime_r50")]
     #[cfg_attr(gnu_time_bits64, link_name = "__gmtime64_r")]
     #[cfg_attr(not(musl32_time64), allow(deprecated))]
-    #[cfg_attr(musl32_time64, link_name = "__gmtime64_r")]
+    #[cfg_attr(musl_redir_time64, link_name = "__gmtime64_r")]
     pub fn gmtime_r(time_p: *const time_t, result: *mut tm) -> *mut tm;
     #[cfg_attr(target_os = "netbsd", link_name = "__localtime_r50")]
     #[cfg_attr(gnu_time_bits64, link_name = "__localtime64_r")]
     #[cfg_attr(not(musl32_time64), allow(deprecated))]
-    #[cfg_attr(musl32_time64, link_name = "__localtime64_r")]
+    #[cfg_attr(musl_redir_time64, link_name = "__localtime64_r")]
     pub fn localtime_r(time_p: *const time_t, result: *mut tm) -> *mut tm;
     #[cfg_attr(
         all(target_os = "macos", target_arch = "x86"),
         link_name = "mktime$UNIX2003"
     )]
     #[cfg_attr(target_os = "netbsd", link_name = "__mktime50")]
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__mktime64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__mktime64")]
     #[cfg_attr(not(musl32_time64), allow(deprecated))]
     pub fn mktime(tm: *mut tm) -> time_t;
     #[cfg_attr(target_os = "netbsd", link_name = "__time50")]
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__time64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__time64")]
     #[cfg_attr(not(musl32_time64), allow(deprecated))]
     pub fn time(time: *mut time_t) -> time_t;
     #[cfg_attr(target_os = "netbsd", link_name = "__gmtime50")]
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__gmtime64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__gmtime64")]
     #[cfg_attr(not(musl32_time64), allow(deprecated))]
     pub fn gmtime(time_p: *const time_t) -> *mut tm;
     #[cfg_attr(target_os = "netbsd", link_name = "__locatime50")]
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__localtime64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__localtime64")]
     #[cfg_attr(not(musl32_time64), allow(deprecated))]
     pub fn localtime(time_p: *const time_t) -> *mut tm;
     #[cfg_attr(target_os = "netbsd", link_name = "__difftime50")]
-    #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__difftime64")]
+    #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__difftime64")]
     #[cfg_attr(not(musl32_time64), allow(deprecated))]
     pub fn difftime(time1: time_t, time0: time_t) -> c_double;
     #[cfg(not(target_os = "aix"))]
     #[cfg_attr(target_os = "netbsd", link_name = "__timegm50")]
     #[cfg_attr(gnu_time_bits64, link_name = "__timegm64")]
     #[cfg_attr(not(musl32_time64), allow(deprecated))]
-    #[cfg_attr(musl32_time64, link_name = "__timegm_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__timegm_time64")]
     pub fn timegm(tm: *mut crate::tm) -> time_t;
 
     #[cfg_attr(target_os = "netbsd", link_name = "__mknod50")]
@@ -1547,7 +1547,7 @@ extern "C" {
     #[cfg_attr(target_os = "netbsd", link_name = "__select50")]
     #[cfg_attr(target_os = "aix", link_name = "__fd_select")]
     #[cfg_attr(gnu_time_bits64, link_name = "__select64")]
-    #[cfg_attr(musl32_time64, link_name = "__select_time64")]
+    #[cfg_attr(musl_redir_time64, link_name = "__select_time64")]
     pub fn select(
         nfds: c_int,
         readfds: *mut fd_set,
@@ -2112,7 +2112,7 @@ cfg_if! {
     )))] {
         extern "C" {
             #[cfg_attr(target_os = "netbsd", link_name = "__adjtime50")]
-            #[cfg_attr(any(gnu_time_bits64, musl32_time64), link_name = "__adjtime64")]
+            #[cfg_attr(any(gnu_time_bits64, musl_redir_time64), link_name = "__adjtime64")]
             pub fn adjtime(delta: *const timeval, olddelta: *mut timeval) -> c_int;
         }
     } else if #[cfg(target_os = "solaris")] {
@@ -2290,7 +2290,7 @@ cfg_if! {
             )]
             #[cfg_attr(target_os = "netbsd", link_name = "__pselect50")]
             #[cfg_attr(gnu_time_bits64, link_name = "__pselect64")]
-            #[cfg_attr(musl32_time64, link_name = "__pselect_time64")]
+            #[cfg_attr(musl_redir_time64, link_name = "__pselect_time64")]
             pub fn pselect(
                 nfds: c_int,
                 readfds: *mut fd_set,
