diff --git a/src/lib.rs b/src/lib.rs
index b99357d..0871db2 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -24,19 +24,19 @@ mod mac;
 #[cfg(any(target_os = "macos", target_os = "ios"))]
 use mac as sys;

-#[cfg(target_arch = "wasm32")]
+#[cfg(all(target_arch = "wasm32", not(target_os = "linux")))]
 mod wasm;
-#[cfg(target_arch = "wasm32")]
+#[cfg(all(target_arch = "wasm32", not(target_os = "linux")))]
 use wasm as sys;

 #[cfg(not(any(
     target_os = "windows",
     target_os = "macos", target_os = "ios",
-    target_arch = "wasm32"
+    all(target_arch = "wasm32", not(target_os = "linux"))
 )))]
 mod lin;
 #[cfg(not(any(
     target_os = "windows",
     target_os = "macos", target_os = "ios",
-    target_arch = "wasm32"
+    all(target_arch = "wasm32", not(target_os = "linux"))
 )))]
 use lin as sys;
