--- a/lib/Basic/Targets.cpp	2025-10-19 18:46:52.435577208 +0800
+++ b/lib/Basic/Targets.cpp	2025-10-19 18:46:31.389590411 +0800
@@ -683,7 +683,7 @@
   case llvm::Triple::wasm32:
     if (Triple.getSubArch() != llvm::Triple::NoSubArch ||
         Triple.getVendor() != llvm::Triple::UnknownVendor ||
-        !Triple.isOSBinFormatWasm())
+        (!Triple.isOSBinFormatWasm() && os != llvm::Triple::Linux))
       return nullptr;
     switch (os) {
       case llvm::Triple::WASI:
@@ -692,6 +692,9 @@
       case llvm::Triple::Emscripten:
       return std::make_unique<EmscriptenTargetInfo<WebAssembly32TargetInfo>>(
           Triple, Opts);
+      case llvm::Triple::Linux:
+      return std::make_unique<LinuxTargetInfo<WebAssembly32TargetInfo>>(Triple,
+                                                                        Opts);
       case llvm::Triple::UnknownOS:
       return std::make_unique<WebAssemblyOSTargetInfo<WebAssembly32TargetInfo>>(
           Triple, Opts);
@@ -701,7 +704,7 @@
   case llvm::Triple::wasm64:
     if (Triple.getSubArch() != llvm::Triple::NoSubArch ||
         Triple.getVendor() != llvm::Triple::UnknownVendor ||
-        !Triple.isOSBinFormatWasm())
+        (!Triple.isOSBinFormatWasm() && os != llvm::Triple::Linux))
       return nullptr;
     switch (os) {
       case llvm::Triple::WASI:
@@ -710,6 +713,9 @@
       case llvm::Triple::Emscripten:
       return std::make_unique<EmscriptenTargetInfo<WebAssembly64TargetInfo>>(
           Triple, Opts);
+      case llvm::Triple::Linux:
+      return std::make_unique<LinuxTargetInfo<WebAssembly64TargetInfo>>(Triple,
+                                                                        Opts);
       case llvm::Triple::UnknownOS:
       return std::make_unique<WebAssemblyOSTargetInfo<WebAssembly64TargetInfo>>(
           Triple, Opts);
--- a/lib/Driver/ToolChains/WebAssembly.cpp	2025-10-19 18:46:52.442577536 +0800
+++ b/lib/Driver/ToolChains/WebAssembly.cpp	2025-10-19 18:46:39.912990049 +0800
@@ -98,36 +98,40 @@
   const char *Crt1;
   const char *Entry = nullptr;
 
-  // When -shared is specified, use the reactor exec model unless
-  // specified otherwise.
-  if (Args.hasArg(options::OPT_shared))
-    IsCommand = false;
-
-  if (const Arg *A = Args.getLastArg(options::OPT_mexec_model_EQ)) {
-    StringRef CM = A->getValue();
-    if (CM == "command") {
-      IsCommand = true;
-    } else if (CM == "reactor") {
+  if (ToolChain.getTriple().isOSLinux()) {
+    Crt1 = "crt1.o";
+  } else {
+    // When -shared is specified, use the reactor exec model unless
+    // specified otherwise.
+    if (Args.hasArg(options::OPT_shared))
       IsCommand = false;
-    } else {
-      ToolChain.getDriver().Diag(diag::err_drv_invalid_argument_to_option)
-          << CM << A->getOption().getName();
+
+    if (const Arg *A = Args.getLastArg(options::OPT_mexec_model_EQ)) {
+      StringRef CM = A->getValue();
+      if (CM == "command") {
+        IsCommand = true;
+      } else if (CM == "reactor") {
+        IsCommand = false;
+      } else {
+        ToolChain.getDriver().Diag(diag::err_drv_invalid_argument_to_option)
+            << CM << A->getOption().getName();
+      }
     }
-  }
 
-  if (IsCommand) {
-    // If crt1-command.o exists, it supports new-style commands, so use it.
-    // Otherwise, use the old crt1.o. This is a temporary transition measure.
-    // Once WASI libc no longer needs to support LLVM versions which lack
-    // support for new-style command, it can make crt1.o the same as
-    // crt1-command.o. And once LLVM no longer needs to support WASI libc
-    // versions before that, it can switch to using crt1-command.o.
-    Crt1 = "crt1.o";
-    if (ToolChain.GetFilePath("crt1-command.o") != "crt1-command.o")
-      Crt1 = "crt1-command.o";
-  } else {
-    Crt1 = "crt1-reactor.o";
-    Entry = "_initialize";
+    if (IsCommand) {
+      // If crt1-command.o exists, it supports new-style commands, so use it.
+      // Otherwise, use the old crt1.o. This is a temporary transition measure.
+      // Once WASI libc no longer needs to support LLVM versions which lack
+      // support for new-style command, it can make crt1.o the same as
+      // crt1-command.o. And once LLVM no longer needs to support WASI libc
+      // versions before that, it can switch to using crt1-command.o.
+      Crt1 = "crt1.o";
+      if (ToolChain.GetFilePath("crt1-command.o") != "crt1-command.o")
+        Crt1 = "crt1-command.o";
+    } else {
+      Crt1 = "crt1-reactor.o";
+      Entry = "_initialize";
+    }
   }
 
   if (!Args.hasArg(options::OPT_nostdlib, options::OPT_nostartfiles))
