diff --git a/std/start.zig b/std/start.zig
index 44fef31..bc4530c 100644
--- a/std/start.zig
+++ b/std/start.zig
@@ -644,7 +644,10 @@ fn main(c_argc: c_int, c_argv: [*][*:0]c_char, c_envp: [*:null]?[*:0]c_char) ca

     switch (builtin.os.tag) {
-        .linux => {
+        .linux => if (!(is_wasm and builtin.target.isMuslLibC())) {
+            // A musl-hosted Wasm executable has no ELF image or program-header
+            // auxiliary-vector entries. musl has already initialized the Wasm
+            // TLS image before invoking this function.
             const at_phdr = std.c.getauxval(elf.AT_PHDR);
             const at_phnum = std.c.getauxval(elf.AT_PHNUM);
             const phdrs = (@as([*]elf.Phdr, @ptrFromInt(at_phdr)))[0..at_phnum];
